[2nd try][Traceable FSDP2] Allow tracing through FSDP2 impl in trace_rules.py

yf225 opened the pull request:
The previous PR https://github.com/pytorch/pytorch/pull/133532 caused stuck compilation issue on internal models. In this 2nd attempt PR, we gate the trace_rules.py changes with `if not torch._dynamo.config.skip_fsdp_hooks:`, so that they don't take effect for current graph-break FSDP2 (which relies on the default config value `skip_fsdp_hooks=True`), and will only take effect when we are using Traceable FSDP2 (in which case the user needs to proactively set `skip_fsdp_hooks=False`).

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #134539



cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

yf225 committed:
[2nd try][Traceable FSDP2] Allow tracing through FSDP2 impl in trace_rules.py

[ghstack-poisoned]

ckluk2 commented:
LGTM

ckluk2 reviewed (commented):
LGTM

yf225 commented:
@pytorchbot merge

yf225 commented:
@pytorchbot merge

yf225 commented:
@pytorchbot merge -f "unrelated rocm failure"

yf225 committed:
Update on "[2nd try][Traceable FSDP2] Allow tracing through FSDP2 impl in trace_rules.py"

The previous PR https://github.com/pytorch/pytorch/pull/133532 caused stuck compilation issue on internal models. In this 2nd attempt PR, we gate the trace_rules.py changes with `if not torch._dynamo.config.skip_fsdp_hooks:`, so that they don't take effect for current graph-break FSDP2 (which relies on the default config value `skip_fsdp_hooks=True`), and will only take effect when we are using Traceable FSDP2 (in which case the user needs to proactively set `skip_fsdp_hooks=False`).




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

yf225 commented:
@pytorchbot merge

ZainRizvi commented:
@pytorchbot revert -c ghfirst "Sorry, I had to revert this in order to revert another PR"

ZainRizvi commented:
@pytorchbot revert -c ghfirst -m "Sorry, I had to revert this in order to revert another PR"

yf225 commented:
@pytorchbot rebase

yf225 committed:
Update on "[2nd try][Traceable FSDP2] Allow tracing through FSDP2 impl in trace_rules.py"

The previous PR https://github.com/pytorch/pytorch/pull/133532 caused stuck compilation issue on internal models. In this 2nd attempt PR, we gate the trace_rules.py changes with `if not torch._dynamo.config.skip_fsdp_hooks:`, so that they don't take effect for current graph-break FSDP2 (which relies on the default config value `skip_fsdp_hooks=True`), and will only take effect when we are using Traceable FSDP2 (in which case the user needs to proactively set `skip_fsdp_hooks=False`).




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

yf225 commented:
@pytorchbot merge
